src/DIRAC/ConfigurationSystem/Client/VOMS2CSSynchronizer.py

```diff
@@ -87,8 +87,15 @@ def _getUserNameFromDN(dn, vo):
                     return nname
                 else:
                     robot = False
+                    # only pop if the remains are sufficient (i.e. not just digits)
                     if names[0].lower().startswith("robot"):
-                        names.pop(0)
+                        nameok = False
+                        if len(names) > 1:
+                            for name in names[1:]:
+                                if not name.isdigit():
+                                    nameok = True
+                        if nameok:
+                            names.pop(0)
```

Comment:
Is the logic here intended to be this (I think it's equivalent to what is currently written):
```suggestion
                        for name in names[1:]:
                            if not name.isdigit():
                                # Remove the "robot.*" prefix
                                names.pop(0)
                                break
```